File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let mp={};
    for(let i=0;i<strs.length;i++){
        let str=strs[i].split('').sort().join('');
        if(!mp[str]){
            mp[str]=[strs[i]];
        }else mp[str].push(strs[i]);
    }
    return Object.values(mp);
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
     def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mapp=defaultdict(list)
        ans=[]
        for s in strs:
            key="".join(sorted(s))
            mapp[key].append(s)
    
        return list(mapp.values())
        